code/obj/item/Bamboo_spike_trap.dm

```diff
@@ -0,0 +1,207 @@
+			else
+				boutput(User, "<span class='alert'>A trap is already placed here!</span>")
+		..()
+
+
+	Crossed(atom/movable/AM as mob|obj)
+		..()
+		if (src.armed && ishuman(AM))
+			var/mob/living/carbon/human/victim = AM
+			//crawling or just walking between the sticks is a viable counter
+			//getting thrown at the trap has a different effect we want to check seperately
+			if(victim.lying || victim.throwing || !src.checkRun(victim))
+				return
+			//If any checks failed, well, you step into the trap
+			victim.visible_message("<span class='alert'><B>[victim] steps into the spike trap!</B></span>",\
+			"<span class='alert'><B>You step into the spike trap!</B></span>")
+			step_on(victim)
+
+	proc/crash_into(mob/living/carbon/human/victim as mob)
+		if (!src || !victim || !src.armed)
+			return
+		logTheThing(LOG_COMBAT, victim, "crashed into [src] at [log_loc(src)].")
+		victim.changeStatus("stunned", src.crashed_stun)
+		victim.force_laydown_standup()
+		src.trap_damage(victim, src.crashed_force)
+		playsound(victim.loc, 'sound/impact_sounds/Generic_Hit_Heavy_1.ogg', 80, 1)
+		victim.UpdateDamageIcon()
+
+
+	proc/step_on(mob/living/carbon/human/victim as mob)
+		if (!src || !victim || !src.armed)
+			return
+		logTheThing(LOG_COMBAT, victim, "stepped into [src] at [log_loc(src)].")
+		victim.changeStatus("weakened", src.armed_weakened)
+		victim.force_laydown_standup()
+		src.trap_damage(victim, src.armed_force)
+		playsound(victim.loc, 'sound/impact_sounds/Flesh_stab_1.ogg', 80, 1)
+		victim.UpdateDamageIcon()
+
+	proc/trap_damage(mob/living/carbon/human/victim as mob, damage)
+		if (!src || !victim)
+			return
+		var/target = "All"
+		if (victim.organHolder[src.target_zone])
+			target = src.target_zone
+		// we need this to calculate how much chems get transfered
+		// This means damage against the zone, reduced by melee protection, multiplied by transfer multiplier and then rounded
+		var/injected_amount = max(0, round((damage - victim.get_melee_protection(target, DAMAGE_STAB))*src.transfer_multiplier))
+		victim.TakeDamageAccountArmor(target, damage, 0, 0, DAMAGE_STAB)
+		// If injected_amount is greater than 0 and there are reagents in the trap, inject the victim
+		if (src.reagents && src.reagents.total_volume && injected_amount > 0)
+			logTheThing(LOG_COMBAT, src, "injected [victim] at [log_loc(src)] with [min(injected_amount, src.reagents.total_volume)]u of reagents.")
+			src.reagents.trans_to(victim, injected_amount)
+
+	//This is copied out of runetrap.dm and modified to work with like the banana peel
+	//So, if you can slip over a banana peel (it has ignore_actual_delay checked), you should normally get impaled by the trap
+	proc/checkRun(var/mob/M)	//If we are above walking speed, this triggers
+		if(!M)
+			return
+		var/slip_delay = BASE_SPEED_SUSTAINED + WALK_DELAY_ADD
+		var/movement_delay_real = max(M.movement_delay(get_step(M,M.move_dir), 0),world.tick_lag)
+		if (movement_delay_real < slip_delay)
+			return TRUE
```

Comment:
this should probably just be a `mob` proc?

Reply:
A check like that is in multiple places. Like runtrap.dm has one version that is copied from mob/proc/slip and this one uses a different version that is also copied from there. Turning it in it's own mob/proc is probably a good idea and makes it more flexible for other functions as well

Reply:
While looking at that, it actually doesnt catch when someone runs... will change that real quick...

Reply:
the whole thing was moved to be a mob proc.